front page: Overhaul content (#373)
cockpit-project/cockpit-project.github.io: index.md

layout: essential
class: front-page
---
{% capture newline %}
{% endcapture %}

{% capture screenshots %}
login-opt.png: Cockpit's log in prompt (on Fedora 34)
debian-in-windows-edge.jpg: Cockpit works where you are (Pictured: Connecting to Debian server from Microsoft Edge on Windows 10)
journal.png: View, filter, and search system logs
accounts.png: Edit accounts

> firewall-rhel.png: Edit the firewall with ease (Pictured: Cockpit Web Console on Red Hat Enterprise Linux, connected from Fedora 34 Workstation)

Reviewer: This screenshot has a *lot* of desktop background, was that intended? It makes the fonts quite a bit blurry.

Author: Yeah, I threw in a couple full screenshots and some cropped to browser UI to show context. I wanted to switch it up and show that it runs in Windows and Linux (and to show off that Linux actually has desktops UIs too; it's not just for servers).

They're _heavily optimized_ for file size. And they're not blurry at all, unless the browser scales it down. It's possible to middle click, control-click, or right click (and view link) to open in tabs and see the file 1:1 in the browser as well.

Author: We could also swap out the page shown if highlighting the firewall makes more sense.

{% capture screenshots_html %}
<div class="screenshots">{%
for screenshot in screenshots
  %}{%
  assign filename = screenshot | split: ":" | first | strip
    | prepend: "/images/screenshot/"
    | prepend: site.baseurl
  %}{%
  assign text = screenshot | replace_first: ":", "©"
    | split: "©" | last | strip
  %}{%
  unless text == ""
    %}<a class="screenshot zoom"
    href="{{ filename }}"><img src="{{ filename }}" 
    title="{{ text }}" alt="{{ text }}"></a>{% 
  endunless
  %}{%
endfor
%}</div>
{% endcapture %}

{% comment %}
##### Content #####
First we set up & capture the content, then we render it in the scaffolding below.
{% endcomment %}

{% capture intro-lede %}
The easy-to-use, integrated, glanceable, and open web-based interface for your servers
{% endcapture %}

{% capture intro-text %}
## Introducing Cockpit

Cockpit is a web-based graphical interface for servers, intended for everyone, especially those who are:

{:.audience-list}
- **new to Linux** (including Windows admins)
- **familiar with Linux** and want an easy, graphical way to administer servers
- **expert admins** who mainly use other tools but want an overview on individual systems

> Thanks to Cockpit intentionally using system APIs and commands, a whole team of admins can manage a system in the way they prefer, including non-Cockpit commands and utilities right alongside Cockpit.

Reviewer: ```suggestion
Thanks to Cockpit intentionally using system APIs and commands, a whole team of admins can manage a system in the way they prefer, including the command line and utilities right alongside Cockpit.
```
I find that "non-Cockpit commands" sounds confusing.

class: front-page
---
{% include page_header.html %}
{% capture newline %}
{% endcapture %}

{% comment %}
##### Content #####
First we set up & capture the content, then we render it in the scaffolding below.
{% endcomment %}

{% capture intro-lede %}
The easy-to-use, integrated, glanceable, and open web-based interface for your servers
{% endcapture %}

{% capture intro-text %}
## Introducing Cockpit

Cockpit is a web-based graphical interface for servers, intended for everyone, especially those who are:

{:.audience-list}
- **new to Linux** (including Windows admins)
- **familiar with Linux** and want an easy, graphical way to administer servers
- **expert admins** who mainly use other tools but want an overview on individual systems

> Thanks to Cockpit intentionally using system APIs and commands, a whole team of admins can manage a system in the way they prefer, including non-Cockpit commands and utilities right alongside Cockpit.

Reviewer: This feels a bit odd to me. How about this:
```suggestion
Cockpit uses the official system APIs and commands and does not interfere with other tools. Hence, administrators can manage a system in the way they prefer, including non-Cockpit commands and utilities right alongside Cockpit.
```

Author: Thanks for the suggestion! I've been so close to the trees, I haven't seen all of the forest, so to speak.

- Extend Cockpit's functionality by installing a growing list of apps and add-ons
- [Write your own custom modules]({{ site.baseurl }}/blog/cockpit-starter-kit.html) to make Cockpit do anything you want

*[RAID]: Redundant Array of Inexpensive Disks
*[LUKS]: Linux Unified Key Setup (encryption)

Also troubleshoot and fix pesky problems with ease:

{:.using-cockpit}
- Diagnose network issues
- Spot and react to misbehaving virtual machines
- Examine SELinux logs and fix common violations in a click
- Inspect detailed metrics that correlate CPU load, memory usage, network activity, and storage performance with the system's journal

More features appear in Cockpit every release.

### Designed & tested

Cockpit's design keeps your goals in mind.  We test Cockpit with usability studies to make it work the way you'd expect and adjust accordingly. As a result, Cockpit gets easier to use all the time.

All code changes have tests which must pass before merging, to ensure stability.

### Free & free

Cockpit is free to use and [available under the GNU LGPL](https://github.com/cockpit-project/cockpit/blob/master/COPYING).

### Cockpit works (nearly) everywhere

> You can install Cockpit on the major distributions, including:

Reviewer: Let's make this a link to running.html. I know we have several already, but it can't hurt?

{:.distro-logos}
{%
  for distro in site.data.distros
%}- [![{{ distro.name }}](/images/site/os-{{ distro.first}}.svg)]({{ site.baseurl }}/running.html#{{ distro.first }})
{% endfor %}

Once Cockpit is up and running, you can access systems from all major web browsers on any operating system (including Windows, MacOS, and Android).

## Release schedule

Cockpit has a time-based release cadence, with new versions appearing every two weeks.

## Get started

{:style="text-decoration-skip-ink:none"}

> After [installing and enabling Cockpit]({{ site.baseurl }}/running.html), visit port 9090 on your server (for example: <https://localhost:9090/> in a browser on the same machine as Cockpit).

Reviewer: For localhost we could even use http://. That would avoid the pesky self-signed certificate problem, and for localhost you don't gain anything with encryption.

Of course that does not translate to remote hosts, but there cockpit automatically redirects http to https anyway. So maybe overall this would be a better choice? This could even format the link to not show the schema at all, like 

    After..., [visit port 9090 on your machine](http://localhost:9090), or on your remote server.

?

Author: I had the URL exposed to demonstrate that it would go to one's local machine. If it's hidden away, it's just a broken link for those who aren't using Cockpit.

Generally, I don't like exposing URLs as linked text, but thought it made sense in this case.

We _could_ use a hybrid approach and make the link contain the text `localhost:9090` (not showing `http`/`https` and the `://`), which also happens to work in browsers just fine.

Reviewer: Good point about being explicit. Then I think it's not actually bad to show http://localhost:9090 explicitly.

[![Docker screenshot]({{ site.baseurl }}/images/site/screenshot-docker.png)]({{ site.baseurl }}/images/site/screenshot-docker.png){:.screenshot.zoom}
{% endcapture %}

{% capture intro-text %}
## Introducing Cockpit

Cockpit is a web-based graphical interface for servers and is intended for everyone, especially those who are:

{:.audience-list}
- **new to Linux** (including Windows admins)
- **familiar with Linux** and want an easy, graphical way to administer servers
- **expert admins** who use other tools and want to quickly check-in on systems

## Cockpit is (nearly) everywhere

You can install Cockpit on the major distributions, including:

{:.distro-logos}
{%
  for distro in site.data.distros
%}- [![{{ distro.name }}](/images/site/os-{{ distro.first}}.svg)]({{ site.baseurl }}/running.html#{{ distro.first }})
{% endfor %}

> Once Cockpit is up and running, you can access systems from all major web browsers on operating systems (and, yes, this even includes Windows and macOS).

Reviewer: ```suggestion
Once Cockpit is up and running, you can access systems from all major web browsers on any operating system (including Windows, MacOS, and Android).
```

I forget, which one was the problematic one with the self-signed certs? iOS or MacOS?

Author: iOS is the problem. It's due to certs and only having one web browser, Safari (the others are just reskinned UIs around Safari's WebKit), so any isn't quite right. I had "all" earlier but removed it.

All major web browsers → Safari does actually work on macOS, so it's not a problem due to WebKit (or even Safari's version of it), but the differing implementation on iOS.

## Simple to use

After installing and enabling Cockpit, accessing Cockpit is usually as easy as visiting port 9090 on your server (for example: <https://localhost:9090/> in a browser on the same machine as Cockpit).

> But don't worry&mdash;Cockpit itself doesn't eat resources or even run in the background when you're not using it. It only starts on demand, thanks to using a systemd socket.

Reviewer: ```suggestion
But don't worry&mdash;Cockpit itself doesn't eat resources or even run in the background when you're not using it. It only starts on demand, thanks to using systemd socket activation.`
```

## Compatible with your existing workflows

> Have a favorite app or command line tool that you use on your servers? You can keep using it. Cockpit uses the same system tooling you'd use from the command line. You can switch back and forth between Cockpit and whatever else you like. (And you can even use some of those tools from within Cockpit's built-in terminal.)

Reviewer: ```suggestion
Have a favorite app or command line tool that you use on your servers? You can keep using it. Cockpit uses the same system tooling you would use from the command line. You can switch back and forth between Cockpit and whatever else you like. Cockpit even has  built-in terminal, which is useful  when you connect from a non-Linux device.
```
